### PR TITLE
update to latest version of istio-api repo

### PR DESCRIPTION
**Pull request by ZackButcher**
**What this PR does / why we need it**: Update to the latest version of istio/api, which refactors where the various proxy related config protos live.

Note this only affects //pilot.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pilot/adapter/config/ingress/controller_test.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/client-go/kubernetes/fake"
 
-	proxyconfig "istio.io/api/proxy/v1/config"
+	proxyconfig "istio.io/api/routing/v1alpha1"
```
Comment (ayj) [inline pilot/adapter/config/ingress/controller_test.go]: `routingconfig` or maybe even `routing` to match the unaliased package name?

Comment (ZackButcher) [reply pilot/adapter/config/ingress/controller_test.go]: Done, went with `routing`.

**Comment by ZackButcher** [comment]
Not at all clear to me why prow presubmit is failing - it logs that the build failed, but I can't find any indication of failure from bazel in the logs it produces (and I can't repro locally).

**Comment by ayj** [comment]
Looks like `bin/fmt.sh` check failed.

```
W1128 22:03:53.982] + git status
W1128 22:03:53.990] + die 'Repo has unstaged changes. Re-run ./bin/fmt.sh'
W1128 22:03:53.991] + echo 'Repo has unstaged changes. Re-run ./bin/fmt.sh'
W1128 22:03:53.991] + exit 1
I1128 22:03:53.991] On branch test
I1128 22:03:53.991] Changes not staged for commit:
I1128 22:03:53.991]   (use "git add <file>..." to update what will be committed)
I1128 22:03:53.992]   (use "git checkout -- <file>..." to discard changes in working directory)
I1128 22:03:53.992] 
I1128 22:03:53.992] 	modified:   Gopkg.lock
I1128 22:03:53.992] 	modified:   generated_files
I1128 22:03:53.992] 
I1128 22:03:53.992] no changes added to commit (use "git add" and/or "git commit -a")
I1128 22:03:53.993] Repo has unstaged changes. Re-run ./bin/fmt.sh
E1128 22:03:53.993] Build failed
I1128 22:03:53.993] process 513 exited with code 1 after 8.0m
E1128 22:03:53.993] FAIL: istio-presubmit
```

**Comment by ZackButcher** [comment]
Thanks, I was searching around for bazel failures and missed that.

**Comment by ZackButcher** [comment]
Presubmit appears to be failing because Gopkg.lock and generated_files are being changed? I don't touch them in this PR.

**Comment by ayj** [comment]
`run_or_die_on_change ./bin/fmt.sh` probably needs to run before `dep ensure` and `bin/init.sh`

/cc @jmuk

**Comment by ZackButcher** [comment]
Why are we running any go commands at all as part of our presubmit? We have a single build system, bazel. This PR works under bazel. There is no reason we should be using any other tooling.

**Comment by jmuk** [comment]
fmt.sh relies on bazel, needs to run *after* init.sh. Didn't notice about Gopkg.lock changes can happen, but that file needs to be updated anyways, doesn't they?

**Pull request merged**